client.go

```diff
@@ -178,6 +179,28 @@ func (c *Client) ExchangeContext(ctx context.Context, m *Msg, a string) (
 	return r, rtt, nil
 }
 
+// Client.DialTimeout acts like DialTimeout, but uses the client's configuration
+func (c *Client) DialTimeoutFunc(network, address string) (conn *Conn, err error) {
+	dialer := net.Dialer{Timeout: c.dialTimeout(), LocalAddr: c.LocalAddr}
+
+	conn = new(Conn)
+	if conn.Conn, err = dialer.Dial(network, address); err != nil {
+		return nil, err
+	}
+	return conn, nil
+}
+
+// Client.DialTimeoutWithTLS acts like DialTimeoutWithTLS, but uses the client's configuration
+func (c *Client) DialTimeoutWithTLS(network, address string) (conn *Conn, err error) {
+	dialer := net.Dialer{Timeout: c.dialTimeout(), LocalAddr: c.LocalAddr}
+
+	conn = new(Conn)
+	if conn.Conn, err = tls.DialWithDialer(&dialer, network, address, c.TLSConfig); err != nil {
+		return nil, err
+	}
+	return conn, nil
+}
+
```

Comment:
I generally agree with you, but looking at this file's layout, I see, in order:
1) type definitions
2) various `Exchange*` functions
3) all the struct methods of `Client`
4) the various `Dial*` functions (non-struct methods)

If I move those two new `Client` methods below the referenced functions (i.e. after `4`) we would fragment the `Client` methods (`3`) along the file, making it less natural to follow the file content.

What about instead moving `Dial` and `DialTimeout` above, i.e. before any `Client` method is defined? CC @miekg as this would touch other parts of the file